.github/workflows/documentation.yml

```diff
@@ -2,7 +2,9 @@ name: deploy documentation
 
 on:
   push:
-    branches: [master]
+    branches:
+      - master
+  pull_request:
```

Comment:
we should build the doc on pull requests too. just not deploy it - see the if statement in the deploy step